misc/peer-store/src/behaviour.rs

```diff
@@ -149,13 +120,8 @@ where
         cx: &mut std::task::Context<'_>,
     ) -> std::task::Poll<libp2p_swarm::ToSwarm<Self::ToSwarm, libp2p_swarm::THandlerInEvent<Self>>>
     {
-        if let Some(ev) = self.store.poll(cx) {
-            self.handle_store_event(ev);
-        };
-
-        if let Some(ev) = self.pending_events.pop_front() {
-            return Poll::Ready(libp2p_swarm::ToSwarm::GenerateEvent(ev));
-        }
-        Poll::Pending
```

Comment:
Drive-by simplification. There is not need for the extra `pending_events` queue, the behavior can directly return the store events.